packages/auth-ts/rollup.config.js

```diff
@@ -53,6 +53,19 @@ const es5Builds = [
     output: [{ file: pkg.main, format: 'cjs', sourcemap: true }],
     plugins: es5BuildPlugins,
     external: id => deps.some(dep => id === dep || id.startsWith(`${dep}/`))
+  },
+  /**
+   * App React Native Builds
+   */
+  {
+    input: 'src/legacy_polyfill.rn.ts',
```

Comment:
old SDK only had one file for everything, so we should only have one legacy polyfill that includes everything

Reply:
Haven't we established that the old SDK did things wrong? I don't think that's good justification for making one entrypoint -- especially since the configuration of the other rollup files (app, firebase) do this as well for tree-shaking purposes. If you would like me to only have one file then we are going to have to access closure to access the dependency injected libraries we need (in extended namespace firebase.INTERNAL.reactNative.AsyncStorage). Is that what you want?

Reply:
for the polyfill we need to support the old API surface which was "wrong", the new SDK API provides multiple entry points

Reply:
In the firebase polyfill/compat layer, I want to avoid recreating the legacy internal APIs(used only by auth today) unless absolutely necessary, so it would be great if you can handle the dependencies in auth itself.

For a rn specific entry point, I don't feel strongly one way or the other. The plan is to support polyfills for 1 major release to allow people to upgrade to the modular SDKs piecemeal, so they doesn't have to be super optimized.

Reply:
So I tried to bundle `@react-native-community/async-storage` and failed because its main field points to a [Flow](https://flow.org/en/) file. 
You might get it to work by compiling the Flow file first, but It clearly says to me that user shouldn't bundle `@react-native-community/async-storage`.

Reply:
For the compat layer we will be using the old 'react-native' dependency as before. The new PR (https://github.com/firebase/firebase-js-sdk/pull/2947) has this.